apollo-ios-pagination/Sources/ApolloPagination/GraphQLQueryPager.swift

```diff
@@ -2,204 +2,342 @@ import Apollo
+      let initialPublisher = await pager.$initialPageResult
+      varMapPublisher.combineLatest(initialPublisher).sink { [weak self] _ in
+        guard let self else { return }
+        Task {
+          let value = await self.pager.pageTransformation()
+          self.canLoadNextSubject.send(value?.canLoadMore ?? false)
+        }
+      }.store(in: &cancellables)
+    }
+  }
 
-  public typealias Output = (InitialQuery.Data, [PaginatedQuery.Data], UpdateSource)
+  deinit {
+    cancellables.forEach { $0.cancel() }
+  }
+
+  public init(pager: Actor) {
+    self.pager = pager
+  }
+
+  public func subscribe(onUpdate: @MainActor @escaping (Result<Output, Error>) -> Void) {
+    Task {
+      await pager.subscribe(onUpdate: onUpdate)
+        .store(in: &cancellables)
+    }
+  }
+
+  public var canLoadNext: Bool { canLoadNextSubject.value }
 
-  private let client: any ApolloClientProtocol
-  private var firstPageWatcher: GraphQLQueryWatcher<InitialQuery>?
-  private var nextPageWatchers: [GraphQLQueryWatcher<PaginatedQuery>] = []
-  private let initialQuery: InitialQuery
-  let nextPageResolver: (PaginationInfo) -> PaginatedQuery?
-  let extractPageInfo: (PageExtractionData) -> PaginationInfo
-  var currentPageInfo: PaginationInfo? {
-    guard let last = pageOrder.last else {
-      return initialPageResult.flatMap { extractPageInfo(.initial($0)) }
-    }
-    if let data = varMap[last] {
-      return extractPageInfo(.paginated(data))
-    } else if let initialPageResult {
-      return extractPageInfo(.initial(initialPageResult))
-    } else {
-      return nil
+  public func cancel() {
+    Task {
+      await pager.cancel()
     }
   }
 
-  var _subject: CurrentValueSubject<Result<Output, Error>?, Never> = .init(nil)
-  public var subject: AnyPublisher<Result<Output, Error>, Never> { _subject.compactMap({ $0 }).eraseToAnyPublisher() }
-  private var subscribers: [AnyCancellable] = []
+  public func loadMore(
+    cachePolicy: CachePolicy = .fetchIgnoringCacheData,
+    completion: (@MainActor () -> Void)? = nil
+  ) throws {
+    Task {
+      try await pager.loadMore(cachePolicy: cachePolicy)
+      await completion?()
+    }
+  }
 
-  var initialPageResult: InitialQuery.Data?
-  var latest: (InitialQuery.Data, [PaginatedQuery.Data])? {
-    guard let initialPageResult else { return nil }
-    return (initialPageResult, pageOrder.compactMap({ varMap[$0] }))
+  public func refetch(cachePolicy: CachePolicy = .fetchIgnoringCacheData) {
+    Task {
+      await pager.refetch(cachePolicy: cachePolicy)
+    }
   }
 
-  /// Array of page info used to fetch next pages. Maintains an order of values used to fetch each page in a connection.
-  var pageOrder = [AnyHashable]()
+  public func fetch() {
+    Task {
+      await pager.fetch()
+    }
+  }
+}
 
-  /// Maps each query variable set to latest results from internal watchers.
-  var varMap: [AnyHashable: PaginatedQuery.Data] = [:]
+extension GraphQLQueryPager {
+  public actor Actor {
```

Comment:
Should the actor itself really be `public`? Or should you just access it through the `public` methods of the pager itself? If we don't reccomend people to access this directly, let's keep it `internal` so that changes in the future can be made without major version bumps.